materials/section-3/README.md

```diff
@@ -11,17 +11,99 @@
+- Refs do not exist in ClojureScript (but we can just use Atoms instead, which work exactly the same as they do in Clojure)
+- ClojureScript only supports integer and floating point literals that can be mapped to JavaScript primitives.
+  - Therefore, Ratio, BigDecimal & BigInteger do not exist in ClojureScript.
+  - Equality on numbers works as it does in JavaScript, and not in Clojure.
+    - `(= 1.0 1) => true` in ClojureScript, but not in Clojure!
+
+
+### Interop between ClojureScript <-> JavaScript
+
+ClojureScript defines a namespace called `js` that allows us to access JavaScript objects, functions and primitives defined in the global scope.
+
+```clojure
+(def text js/globalName)
+```
+
+If we want to invoke JavaScript functions, we can do so by prefixing the function name with a `.`. We can pass function arguments at the end of each form.
+
+```clojure
+(.getElementById js/document "my-element")
+```
+n
+If we want to access properties of a JavaScript object, we can do so with the `.-` property access syntax.
+
+```clojure
+(.-bar js/foo) ;; JS output: foo.bar;
+```
+
+You can access nested properties using the `..` access syntax.
+
+```clojure
+(.. js/foo -bar -baz) ;; JS output: foo.bar.baz;
+```
+
+We can create JavaScript objects in ClojureScript using the `js-obj` macro:
+
+```clojure
+(js-obj "foo" "bar") ;; creates { foo: "bar"}
+```
+
+This only allows you to use basic data literals as values, though. Any ClojureScript data structures (such as vectors, hash sets etc) won't be changed. To recursively change values into JavaScript, we can use the `clj->js` function:
+
+```clojure
+(clj->js :a [1 2 3] :b #{"wow" "what" "fun"})
+```
+
+This will correspond to
+
+```javascript
+{ 
+  "a": [1, 2, 3],
+  "b": ["wow", "what", "fun"]
+}
+```
+
+If we want to do the same, but in reverse (recursively transform JavaScript into ClojureScript maps), we can do this in such a way using the `js->clj` function:
+
+```clojure
+(def js-array (js-obj "foo" "bar" "awesome" true))
+(js->clj js-array :keywordize-keys true) ;; => {:foo "bar", :awesome true}
+```
+As you can see above, we can add a flag to automatically keywordise the JavaScript object keys, if we would like.
+
+If you'd like to read a little more in depth, I'd recommend turning to [this blog post](https://www.spacjer.com/blog/2014/09/12/clojurescript-javascript-interop/) which provides a great rundown of additional ways you can interact with JS from ClojureScript, and vice-versa.
 
 ---
 
 ## 2. Introduction to Reagent
 
-TBD
-- -> https://reagent-project.github.io/
-- until managing state
+Reagent is a minimalistic interface between ClojureScript and React, and we will be using this today as the base of our frontend. It plays into ClojureScripts strengths by using just ClojureScript data and functions to define React components (rather than JSX sticking out like a sore thumb in the middle of some JavaScript code).
```

Comment:
Is this enough? I feel like since we're just pointing users to the reagent docs, there's not too much else to say here.